legacy/firmware/ethereum.c

```diff
@@ -1042,26 +1052,27 @@ bool ethereum_parse(const char *address, uint8_t pubkeyhash[20]) {
 }
 
 bool ethereum_path_check(uint32_t address_n_count, const uint32_t *address_n,
-                         bool pubkey_export, uint64_t chain) {
+                         bool pubkey_export,
+                         const EthereumDefinitionsDecoded *defs) {
```

Comment:
this is only ever looking at the network part so let's just pass network.

and in fact let's make the argument mandatory and return false if it is NULL

Reply:
9556b8f18d0dab6fc84e20f31bad8ba707737562